JavaScript/Timer.js

```diff
@@ -0,0 +1,63 @@
+//this code is written to operationalize a timer with 3 main functions of Start, Stop and Reset.
+class Clock {
+    /*fourth take */
+
+    constructor(displayElement) {
+        this.display = displayElement;
+        this.state = "initial";
+        this.seconds = 0;
+        this.interval = null;
+    }
+
+    start() {
+        if (this.state === "initial") {
+            this.interval = setInterval(() => this.timer(), 1000);
+            this.state = "running"
+        }
+    }
+
+    stop() {
+        if (this.state ==="running"){
+        clearInterval(this.interval);
+        this.state = "stopped";
+        }
+    }
+
+    reset() {
+        this.state = "initial";
+        clearInterval(this.interval);
+        this.seconds = 0;
+        this.display.innerText = "00:00:00";
+    }
+
+    timer() {
+
+
+        let hrs = Math.floor(this.seconds / 3600);
+        let min = Math.floor((this.seconds - hrs * 3600) / 60);
+        let sec = this.seconds % 60;
+
+        if (sec < 10) sec = "0" + sec;
+        if (min < 10) min = "0" + min;
+        if (hrs < 10) hrs = "0" + hrs;
+
+        this.display.innerText = `${hrs}:${min}:${sec}`;
+        this.seconds++;
```

Comment:
When you pass a method to setInterval() or any other function, it is invoked with the wrong value. 
Code executed by setInterval() runs in a separate execution context than the function from which it was called. As a consequence, the this keyword for the called function is set to the window (or global) object, it is not the same as the this value for the function that called setTimeout. See [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#callbacks) for more details.